Track Buddy/MotionManager.swift

```diff
@@ -66,12 +69,57 @@ class MotionManager: ObservableObject {
     private var recentPoints: Deque<CGPoint> = [] // TODO: dluo- think about thread safety?
     
     func pointPath(atScale factor: CGFloat) -> CGMutablePath {
-        let points = recentPoints.map { CGPoint(x: $0.x * factor, y: $0.y * factor) }
+        let xPoints = recentPoints.map { Double($0.x * factor) }
+        let yPoints = recentPoints.map { Double($0.y * factor) }
+        
+        /* Interpolate points to fill in gaps between rate-limited points to create a smooth tracer line.
+         
+         Using this interpolate-to-smooth approach because generating the line with the raw accelerometer data at 100 Hz
+         can produce visible divergence between the movement of the dot in the graph and the line that is traced under it.
+         
+         */
+        
+        let interpolatedXPoints = interpolate(xPoints)
+        let interpolatedYPoints = interpolate(yPoints)
+        
         let path = CGMutablePath()
-        path.addLines(between: points)
+        guard interpolatedXPoints.count == interpolatedYPoints.count else { return path }
```

Comment:
Would an assertionFailure make sense here? I imagine it's a dev error if the counts don't align

Reply:
Yes, agree